backend/fms_core/viewsets/project.py

```diff
@@ -59,4 +62,30 @@ def summary(self, _request):
             "total_count": Project.objects.count(),
             "open_count": Project.objects.filter(status="Open").count(),
             "closed_count": Project.objects.filter(status="Closed").count(),
-        })
\ No newline at end of file
+        })
+    
+    @action(detail=False, methods=["post"])
+    def add_samples_to_study(self, request, pk=None):
+        sample_ids = request.data.get("sample_ids")
+        select_all = request.data.get("select_all", False)
+        project_id = request.data.get("project_id")
+        study_letter = request.data.get("study_letter")
+        step_order = request.data.get("step_order", None)
+
+        samples = (Sample.objects.filter(derived_by_samples__project=project_id, id__in=sample_ids) if not select_all
+                   else Sample.objects.filter(derived_by_samples__project                   =project_id).exclude(id__in=sample_ids)).all()
```

Comment:
weird spacing in here ... not sure if the line break make much sense other than to accommodate line length limits...